modules/host/update.go

```diff
@@ -74,17 +75,29 @@ func (h *Host) ProcessConsensusChange(cc modules.ConsensusChange) {
 	h.mu.Lock()
 	defer h.mu.Unlock()
 
-	h.blockHeight += types.BlockHeight(len(cc.AppliedBlocks))
-	h.blockHeight -= types.BlockHeight(len(cc.RevertedBlocks))
+	// Adjust the height of the host. The host height is initialized to zero,
+	// but the genesis block is actually height zero. For the genesis block
+	// only, the height will be left at zero.
+	//
+	// Checking the height here eliminates the need to initialize the host to
+	// and underflowed types.BlockHeight.
```

Comment:
this is still done in other modules, right?
kinda ugly how this is handled. It would be nice if we could come up with something cleaner, perhaps by altering the semantics of subscription to treat the genesis block specially.